core/src/main/java/com/novoda/noplayer/player/PlayerFactory.java

```diff
@@ -58,10 +67,10 @@ public Player create(DrmType drmType, DrmHandler drmHandler, boolean downgradeSe
     private Player createPlayerForType(PlayerType playerType, DrmType drmType, DrmHandler drmHandler, boolean downgradeSecureDecoder) {
         switch (playerType) {
             case MEDIA_PLAYER:
-                return mediaPlayerCreator.createMediaPlayer(context);
+                return noPlayerMediaPlayerCreator.createMediaPlayer(context);
```

Comment:
An alternative could have been not having the `InternalCreator` in the creators (therefore you don't need testing) and doing the test for the `player.initialise()` here (which I believe is the bit you wanted to test)